### PR TITLE
Build process modifications

### PR DESCRIPTION
**Pull request by mihai-mihalache**
Added TaskArguments Helper and used it to decouple build process from jQuery
The logic behind this is that the majority of the projects already have their version of jQuery or don't use it at all so if they want to build a release and just use it.
Helper implementation should be final and use cases will be added.

### CODE REVIEW

**Comment by mihai-mihalache** [comment]
NOTE: This does not affect the current build in any way. The --skipjquery flag is strictly optional

File: gulpfile.js [hunk 1]
```diff
@@ -51,14 +78,13 @@ const css_src = [
 const sass_docs_src = './demo.scss';
 
 const js_build_src = './js/src/*.js';
-const js_bundle_src = [
+const js_bundle_src = (args.skipjquery) ? js_build_dest + '*.js' : [
```
Comment (jgolieb) [inline gulpfile.js]: You're actually skipping all of Bootstrap's javascript, not just jQuery.

Comment (jgolieb) [inline gulpfile.js]: You're actually just skipping Bootstrap's javascript, not jQuery.

File: gulpfile.js [hunk 2]
```diff
@@ -20,6 +20,33 @@ const connect = require('gulp-connect');
 const babel = require('gulp-babel');
 const uglify = require('gulp-uglify');
 
+// Helpers
+// Task Arguments
```
Comment (jgolieb) [inline gulpfile.js:24]: Please add a comment describing what this does.

**Review by jgolieb — COMMENTED** [review]
jQuery is not part of Bootstrap's bundle that gets built. It is included in index.html separately before the Bootstrap (or Fluid Bootstrap) bundle, so I'm not sure what this accomplishes. What we probably could do is move our "dependencies" to "peerDependencies" in package.json. If you look at [Bootstrap's own package.json](https://github.com/twbs/bootstrap/blob/v4-dev/package.json), that's where they have jQuery listed.

File: gulpfile.js (hunk 2 above)
Comment (mihai-mihalache) [reply gulpfile.js:24]: Added!

File: gulpfile.js (hunk 1 above)
Comment (mihai-mihalache) [reply gulpfile.js]: Yes. I corrected the flag. The idea is that, at least in my project (SmartSense), I won't be using neither jQuery nor Bootstrap Bundle JS. Being an EmberJS SmartSense will have it's own components that will use only the source style sheets files (.scss) and, evidently, the html markup structure from this project.

Comment (jgolieb) [reply gulpfile.js]: OK, I guess that makes sense then.

**Review by jgolieb — APPROVED** [review]

**Pull request merged**